cogs/music_cog.py

```diff
@@ -84,32 +84,49 @@ async def command_play(self, ctx, *args):
 
     @commands.command(name="pause", help="Pauses the current song")
     async def pause(self, ctx, *args):
-        #TODO
-        print("todo")
+        if self.is_playing:
+            self.is_playing = False
+            self.is_paused = True
+            self.vc.pause()
+        elif self.is_paused:
+            self.vc.resume()
 
     @commands.command(name="resume", aliases=["r"], help="Resumes the current song")
     async def resume(self, ctx, *args):
-        #TODO
-        print("todo")
+        if self.is_paused:
+            self.is_paused = False
+            self.is_playing = True
+            self.vc.resume()
 
     @commands.command(name="skip", aliases=["next", "s"], help="Skips the current song")
     async def skip(self, ctx, *args):
-        #TODO
-        print("todo")
-
+        if self.vc != None and self.vc:
+            self.vc.stop()
+            await self.start_playing(ctx)
 
     @commands.command(name="clear", help="Clears the queue")
-    async def clear(self,ctx, *args):
-        #TODO
-        print("todo")
+    async def clear(self, ctx, *args):
+        if self.vc != None and self.is_playing:
+            self.vc.stop()
+        self.music_queue = []
+        await ctx.send("Queue cleared")
 
     @commands.command(name="queue", aliases=["list"], help="Displays the list of songs in order")
     async def queue(self, ctx, *args):
-        #TODO
-        print("todo")
-
+        queue = ""
+        x = 10 # number of songs in the queue to display
+        for i in range(len(self.music_queue)):
+            if i > x:
+                break
+            queue += self.music_queue[i]['title'] + "\n"
```

Comment:
Would like to see something that adds onto the very end, on a new line, the number of songs remaining in the queue. Helpful for when it cuts off the top 10